src/new.rs

```diff
@@ -527,11 +525,31 @@ pub fn hello_world_test() {{
 }
 
 pub fn create(options: NewOptions, version: &'static str) -> Result<()> {
-    validate_name(&options.name)?;
-    validate_root_folder(&options.name)?;
-    let creator = Creator::new(options, version);
+    let name: String = if options.name.is_empty() {
+        get_foldername(&options.project_root).gleam_expect("unable to get folder name")
```

Comment:
`gleam_expect` is to be used in situtations when something has got horribly an unexpectedly wrong, it indicates a major compiler bug.

Here I think we can expect issues to arise when determining the project name, so instead lets return an error.